packages/discovery-provider/src/utils/web3_provider.py

```diff
@@ -25,21 +27,26 @@ def get_web3(web3endpoint=None):
     if web3:
         return web3
 
-    if not web3endpoint:
+    if web3endpoint:
+        web3 = Web3(HTTPProvider(web3endpoint))
+    else:
+        local_rpc = os.getenv("audius_web3_localhost")
+        local_web3 = Web3(HTTPProvider(local_rpc))
+        gateway_web3 = Web3(HTTPProvider(os.getenv("audius_web3_host")))
         # attempt local rpc, check if healthy
         try:
-            local_rpc = os.getenv("audius_web3_localhost")
-            if requests.get(local_rpc + "/health").status_code == 200:
-                web3endpoint = local_rpc
+            block_diff = math.abs(local_web3.eth.get_block_number() - gateway_web3.eth.get_block_number())
+            resp = requests.get(local_rpc + "/health")
+            if resp.status_code == 200 && block_diff < 10000
```

Comment:
nit pull 10000 as a const at the top

USE_LOCAL_RPC_IF_BEHIND_BLOCK_DIFFERENCE = 10000

or sthg like that 🤷